mountpoint-s3/src/main.rs

```diff
@@ -710,11 +705,9 @@ mod tests {
     #[test_case("s3://test-bucket", false; "not providing bare bucket name")]
     #[test_case("~/mnt", false; "directory name in place of bucket")]
     #[test_case("arn:aws:s3::00000000:accesspoint/s3-bucket-test.mrap", true; "multiregion accesspoint ARN")]
-    #[test_case("arn:aws:s3:::doc-example-bucket", false; "incorrect bucket ARN")]
-    #[test_case("arn:aws:ec2:us-east-1:555555555555:instance/i-b188560f", false; "EC2 instance ARN")]
+    #[test_case("arn:non-aws:s3::bucket-name", false; "non aws ARN")]
```

Comment:
Let's not even assume partition will always start with `aws`. It's not guaranteed even if it is true today.

Can we also keep the bucket ARN case? Even if it succeeds and we just want to let endpoint resolver reject the bucket ARN - let's just be clear that this function won't reject it, and the case description will explain this.